plugins/cost-insights/src/components/AlertInsights/AlertDialog.tsx

```diff
@@ -0,0 +1,228 @@
+    onClose();
+    setDisabled(true);
+  }
+
+  const SnoozeForm: Maybe<AlertForm> = snoozed?.SnoozeForm ?? AlertSnoozeForm;
+  const AcceptForm: Maybe<AlertForm> = accepted?.AcceptForm ?? AlertAcceptForm;
+  const DismissForm: Maybe<AlertForm> =
+    dismissed?.DismissForm ?? AlertDismissForm;
+
+  const isSnoozingEnabled = !!snoozed?.onSnoozed;
+  const isAcceptingEnabled = !!accepted?.onAccepted;
+  const isDismissingEnabled = !!dismissed?.onDismissed;
+
+  const isSnoozeFormDisabled = snoozed?.SnoozeForm === null;
+  const isAcceptFormDisabled = accepted?.AcceptForm === null;
+  const isDismissFormDisabled = dismissed?.DismissForm === null;
+  const isFormDisabled =
+    isSnoozeFormDisabled || isAcceptFormDisabled || isDismissFormDisabled;
+
+  const status = [
+    isSnoozingEnabled,
+    isAcceptingEnabled,
+    isDismissingEnabled,
+  ] as const;
+
+  const [Action, action, actioned] =
+    choose(status, [
+      ['Snooze', 'snooze', 'snoozed'],
+      ['Accept', 'accept', 'accepted'],
+      ['Dismiss', 'dismiss', 'dismissed'],
+    ]) ?? [];
+
+  const [title, subtitle] =
+    choose(status, [
+      [snoozed?.title, snoozed?.subtitle],
+      [accepted?.title, accepted?.subtitle],
+      [dismissed?.title, dismissed?.subtitle],
+    ]) ?? [];
+
+  const TransitionProps = {
+    mountOnEnter: true,
+    unmountOnExit: true,
+    // Wait for child component to mount; avoid recycling refs.
+    onEntered() {
+      if (acceptRef.current) {
+        acceptRef.current.id = DEFAULT_FORM_ID;
+      }
+      if (snoozeRef.current) {
+        snoozeRef.current.id = DEFAULT_FORM_ID;
+      }
+      if (dismissRef.current) {
+        dismissRef.current.id = DEFAULT_FORM_ID;
+      }
+    },
+  };
+
+  return (
+    <Dialog
+      open={open}
+      onClose={onDialogClose}
+      scroll="body"
+      maxWidth="lg"
+      TransitionProps={TransitionProps}
+    >
+      <Box display="flex" justifyContent="flex-end">
+        <IconButton
+          className={classes.icon}
+          disableRipple
+          aria-label="close dialog"
+          onClick={onDialogClose}
+        >
+          <CloseIcon />
+        </IconButton>
+      </Box>
+      <DialogContent className={classes.content}>
+        <Box mb={1.5}>
+          <Typography variant="h5">
+            <b>{Action} this action item?</b>
+          </Typography>
+          <Typography variant="h6" color="textSecondary">
+            <b>
+              This action item will be {actioned} for all of {group}.
+            </b>
+          </Typography>
+        </Box>
+        <Box
+          display="flex"
+          flexDirection="column"
+          bgcolor="alertBackground"
+          p={2}
+          mb={1.5}
+          borderRadius={4}
+        >
+          <Typography>
+            <b>{title}</b>
+          </Typography>
+          <Typography color="textSecondary">{subtitle}</Typography>
+        </Box>
+        {isSnoozingEnabled && !isSnoozeFormDisabled && (
```

Comment:
I find this component a bit strange in duplication and that I could produce a weird multi-form with the right props. It seems like the status duplication could be handled outside and just have the props dictate content:

```
<AlertDialog alert={alert} hook={alert.onSnoozed} action="snooze" />
```

Am I oversimplifying?

Reply:
It appears to be so but those booleans will return false when an alert of a different status is rendered. Did a bit of a refactor which should alleviate some of this complexity.